WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Survey/ProductCreationAISurveyUseCase.swift

```diff
@@ -0,0 +1,39 @@
+import Foundation
+
+/// To determine when to show Product Creation AI survey
+///
+final class ProductCreationAISurveyUseCase {
+    private let defaults: UserDefaults
+    private let analytics: Analytics
+
+    init(defaults: UserDefaults = UserDefaults.standard,
+         analytics: Analytics = ServiceLocator.analytics) {
+        self.defaults = defaults
+        self.analytics = analytics
+    }
+
+    var numberOfTimesAIProductCreated: Int {
+        get {
+            defaults.integer(forKey: UserDefaults.Key.numberOfTimesAIProductCreated.rawValue)
+        }
+        set {
+            defaults.set(newValue, forKey: UserDefaults.Key.numberOfTimesAIProductCreated.rawValue)
+        }
+    }
+
+    /// Returns `true` if it is time to present Product Creation AI survey.
+    ///
+    func shouldShowProductCreationAISurvey() -> Bool {
+        guard numberOfTimesAIProductCreated >= 3 else {
+            return false
+        }
+        return !defaults.bool(forKey: UserDefaults.Key.haveAskedConfirmationToShowProductCreationAISurvey.rawValue)
+    }
+
+    /// Saves that we have asked the user to provide feedback in survey
+    ///
+    func didAskConfirmationToShowProductCreationAISurvey() {
```

Comment:
Nit: can we simplify this too?

```suggestion
    func didSuggestProductCreationAISurvey() {
```

Reply:
Done as part of https://github.com/woocommerce/woocommerce-ios/pull/11390/commits/9421bb3befc4933f13c5140646beaed923277382